File: ex9.py
class Solution:
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        str_x = str(x)

        if str_x[::1] == str_x[::-1]:
            return True
        else:
            return False

sol = Solution()
print(sol.isPalindrome(123))

File: code/ex9.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0)    return 0;
        string s = to_string(x);
        int lo = 0, hi = s.length() - 1;
        while (lo < hi) {
            if (s[lo++] != s[hi--])
                return 0;
        }
        return 1;
    }
};